Reorganize compression docs Part 1 (#965)
timescale/docs: timescaledb/overview/core-concepts/compression/architecture.md

# Compression architecture

> <!-- TODO: Add more to reflect the loads and loads of content on this page-->

Author: ```suggestion
```

Behind the scenes, TimescaleDB actually creates a second, under-the-covers,
hypertable with the compressed data. You shouldn't query this table directly.
Run your queries against the original hypertable, and TimescaleDB finds any
compressed data, decompresses it, and returns it to you.
</highlight>

## Data is segmented and ordered
You can control how TimescaleDB combines rows by using the
`timescaledb.compress_orderby` and `timescaledb.compress_segmentby` parameters.

## Indexes are deleted and created
When you compress a chunk, any indexes on that chunk are deleted, and new
indexes are created.

<highlight type="note">
Your old indexes are deleted because data within array-like structures is not 
indexable in the same way as uncompressed data. For example, before compression,
if you have an index on a column named `data`, the index points to individual
rows with individual values for `data`. After compression, the values for `data`
are now stored in array-like structures, so the index can no longer point to
individual values. Only columns used in `segment_by` can be indexed, since they
are not stored as arrays.
</highlight>

## Additional metadata is stored with table

> 

Reviewer: Can we add a sentence here to avoid the naked heading?

pages:

|time|device_id|cpu|energy_consumption|
|-|-|-|-|
|`<POINTER_TO_DATA>`|`<POINTER_TO_DATA>`|`<POINTER_TO_DATA>`|`<POINTER_TO_DATA>`|

This reduces the amount of data that must be read from disk. When querying a
compressed chunk, the database reads the table of pointers, which is relatively
lightweight. It then follows the pointers to bring in only the needed columns
from disk.

TimescaleDB implements pointers and secondary pages by using PostgreSQL's TOAST
(The Oversized-Attribute Storage Technique) feature. To learn more, see the
[PostgreSQL documentation on TOAST][TOAST].

## Type-specific compression algorithms are used
TimescaleDB uses type-specific compression algorithms to compress each column.
This compresses data more efficiently than generic, type-agnostic algorithms.
Here are the algorithms used for each data type:

|Data type|Compression method|
|-|-|
|Integers and integer-like types, including timestamps|Delta-of-delta and simple-8b with run-length encoding|
|Floats|Gorilla compression|
|Columns with low cardinality (a few possible values that repeat often)|Whole-row dictionary compression and LZ compression|
|Other data types|LZ-based array compression|

> To learn more about the compression algorithms used, see the [compression
> algorithms blog post][compression-algorithms].

Reviewer: ```suggestion
To learn more about the compression algorithms used, see the 
[compression algorithms blog post][compression-algorithms].
```

## Interacting with compressed chunks
In many ways, you interact with compressed chunks just as you interact with
uncompressed chunks. But there are some differences.

### Inserts, updates, and deletes

> Since TimescaleDB 2.3, you can insert data into compressed chunks.

Reviewer: ```suggestion
With TimescaleDB 2.3 and above, you can insert data into compressed chunks.
```

> You cannot currently update or delete data in compressed chunks. Work around

Reviewer: ```suggestion
You can't update or delete data in compressed chunks. Work around
```

this by decompressing the chunk, making your changes, and recompressing. You can

> also delete an entire chunk at once without decompressing it. To learn more, see

Reviewer: ```suggestion
delete an entire chunk at once without decompressing it. For more information, see
```
